recipes-ni/ni-hw-scripts/ni-hw-scripts-common.bb

```diff
@@ -5,55 +5,61 @@ LICENSE = "MIT"
 SECTION = "base"
 
 DEPENDS += "\
-	update-rc.d-native \
+	${@bb.utils.contains('INIT_MANAGER','sysvinit','update-rc.d-native','systemd-systemctl-native',d)} \
 "
 
 SRC_URI += "\
 	file://init.d/ni-rename-ifaces \
 	file://init.d/nisetserialnumber \
+	file://ni-rename-ifaces.service \
+	file://nisetserialnumber.service \
 "
 
 S = "${WORKDIR}"
 
-
-inherit allarch
+inherit allarch systemd
 PACKAGE_ARCH = "all"
 PACKAGES:remove = "${PN}-staticdev ${PN}-dev ${PN}-dbg"
 
 
 do_install () {
-	install -d ${D}${sysconfdir}/init.d/
+	install -d ${D}${script_location}
 
-	install -m 0755 ${S}/init.d/ni-rename-ifaces     ${D}${sysconfdir}/init.d
-	install -m 0755 ${S}/init.d/nisetserialnumber    ${D}${sysconfdir}/init.d
+	install -m 0755 ${S}/init.d/ni-rename-ifaces     ${D}${script_location}
+	install -m 0755 ${S}/init.d/nisetserialnumber    ${D}${script_location}
 }
 
-pkg_postinst:${PN} () {
```

Comment:
It's a shame we have two initscripts in one package otherwise we could've used `INITSCRIPT_NAME` and `INITSCRIPT_PARAMS` instead of the custom `pkg_postinst` and then we wouldn't have had to replace it with `python __anonymous()`.